beacon-chain/state/interfaces.go

```diff
@@ -14,6 +14,18 @@ import (
 	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
 )
 
+type GenesisBeaconState interface {
+	SpecParametersProvider
+	ReadOnlyBeaconState
+	ReadOnlyWithdrawals
+	WriteOnlyBeaconState
+	Copy() BeaconState
+	HashTreeRoot(ctx context.Context) ([32]byte, error)
+	FutureForkStub
+	StateProver
+	Stub()
```

Comment:
I don't know how to make `GenesisBeaconState` and `BeaconState` incompatible otherwise.